### DISCUSSION TITLE
Will opening_hours work on ferry?

### DISCUSSION THREAD

**Discussion by nvcnvn** [valhalla/valhalla#4312] [Q&A] [answered]
https://wiki.openstreetmap.org/wiki/Key:opening_hours

I want to avoid routing to ferry outside opening_hours.
I think this is similar to time-base-turn-restriction which already implemented. If not, can you show me the way how to implement and test it? I think I'm ready for the first C++ contribute xD

**Comment by nilsnolde** [comment, marked as answer]
Yeah that’s a good idea. I don’t think you have to do much for this.. look for “conditional” restrictions in mjolnir/pbfgraphparser.cc. You’d need to extend the logic to look at “opening_hours” tag if there’s a ferry, and set a kTimedAllowed restriction. Testing it we’ll cover on the PR then:)

**Reply by nvcnvn** [reply]
Great! Thanks for the explanation :)
